pints/toy/_stochastic_logistic_model.py

```diff
@@ -0,0 +1,160 @@
+#
+# Stochastic degradation toy model.
+#
+# This file is part of PINTS (https://github.com/pints-team/pints/) which is
+# released under the BSD 3-clause license. See accompanying LICENSE.md for
+# copyright notice and full license details.
+#
+from __future__ import absolute_import, division
+from __future__ import print_function, unicode_literals
+import numpy as np
+from scipy.interpolate import interp1d
+import pints
+
+from . import ToyModel
+
+
+class StochasticLogisticModel(pints.ForwardModel, ToyModel):
+    r"""
+    This model describes the growth of a population of individuals, where the
+    birth rate per capita, initially :math:`b_0`, decreases to :math:`0` as the
+    population size, :math:`\mathcal{C}(t)`, starting from an initial
+    population size, :math:`n_0`, approaches a carrying capacity, :math:`k`.
+    This process follows a rate according to [1]_
+
+    .. math::
+       A \xrightarrow{b_0(1-\frac{\mathcal{C}(t)}{k})} 2A.
+
+    The model is simulated using the Gillespie stochastic simulation algorithm
+    [2]_, [3]_.
+
+    *Extends:* :class:`pints.ForwardModel`, :class:`pints.toy.ToyModel`.
+
+    Parameters
+    ----------
+    initial_molecule_count : float
+        Sets the initial population size :math:`n_0`.
+
+    References
+    ----------
+    .. [1] Simpson, M. et al. 2019. Process noise distinguishes between
+           indistinguishable population dynamics. bioRxiv.
+           https://doi.org/10.1101/533182
+    .. [2] Gillespie, D. 1976. A General Method for Numerically Simulating the
+           Stochastic Time Evolution of Coupled Chemical Reactions.
+           Journal of Computational Physics. 22 (4): 403-434.
+           https://doi.org/10.1016/0021-9991(76)90041-3
+    .. [3] Erban R. et al. 2007. A practical guide to stochastic simulations
+           of reaction-diffusion processes. arXiv.
+           https://arxiv.org/abs/0704.1908v2
+    """
+
+    def __init__(self, initial_molecule_count=50):
+        super(StochasticLogisticModel, self).__init__()
+        self._n0 = float(initial_molecule_count)
+        if self._n0 < 0:
+            raise ValueError('Initial molecule count cannot be negative.')
+
+    def n_parameters(self):
+        """ See :meth:`pints.ForwardModel.n_parameters()`. """
+        return 2
+
+    def simulate_raw(self, parameters):
```

Comment:
Do we want this function to be _simulate_raw?

Reply:
Needs better docstring

Reply:
OK, I will change this to _simulate_raw. And not changing to docstring, because some of the other ones like the logistic model, its _simulate doesn't have a docstring either.